modelbaker/dbconnector/db_connector.py

```diff
@@ -50,25 +51,25 @@ def get_provider_specific_names(self):
             "datasettable_name": self.dataset_table_name,
         }
 
-    def map_data_types(self, data_type):
+    def map_data_types(self, data_type: str) -> str:
         """Map provider date/time types to QGIS date/time types"""
         return None
 
-    def db_or_schema_exists(self):
+    def db_or_schema_exists(self) -> bool:
         """Whether the DB (for GPKG) or schema (for PG) exists or not."""
-        raise NotImplementedError
```

Comment:
This was there to make it clear for subclasses which methods do need to be overridden.

Perhaps we can remove the line as you did and stick now to `@abc.abstractmethod`? See https://stackoverflow.com/a/44316506